src/megaapi_impl.cpp

```diff
@@ -1493,6 +1502,117 @@ char *MegaApiImpl::getBlockedPath()
     sdkMutex.unlock();
     return path;
 }
+
+error MegaApiImpl::backupFolder_sendPendingRequest(MegaRequestPrivate* request) // request created in MegaApiImpl::backupFolder()
+{
+    // validate local path and backup name
+    if (!request->getFile())
+    {
+        return API_EARGS;
+    }
+
+    string localPath(request->getFile());
+    string backupName;
+    if (request->getName())
+    {
+        backupName.assign(request->getName());
+    }
+    else    // get the last leaf of local path
+    {
+        LocalPath p = LocalPath::fromPath(localPath, *client->fsaccess);
+        LocalPath l = p.leafName();
+        backupName = l.toPath(*client->fsaccess);
+        request->setName(backupName.c_str()); // use this in putnodes_result()
+    }
+
+#if defined(_WIN32) && !defined(WINDOWS_PHONE)
+    if (!PathIsRelativeA(localPath.c_str()) && ((localPath.size() < 2) || localPath.compare(0, 2, "\\\\")))
+    {
+        localPath.insert(0, "\\\\?\\");
+        request->setFile(localPath.c_str());
+    }
+#endif
+
+    // get current user
+    User* u = client->ownuser();
+
+    // get handle of remote "My Backups" folder, from user attributes
+    if (!u || !u->isattrvalid(ATTR_MY_BACKUPS_FOLDER)) { return API_EACCESS; }
+    const string* handleContainerStr = u->getattr(ATTR_MY_BACKUPS_FOLDER);
```

Comment:
At what point do we actually create the "My Backups" folder and store its handle in the ATTR_MY_BACKUPS_FOLDER attribute?   I would have thought we would do it here if it didn't exist already?   thanks

Reply:
I don't remember where exactly it was written, but the idea was that the folder at the remote location and user attr would already exist when the user will have the possibility of creating backups.

Reply:
It was agreed with MEGAsync TeamLeader that:

> ...upon creation of a new backup, the app should:
> - Have created the "My Backups" folder first (it needs to be localised) and set the corresponding attribute by calling setMyBackupsFolder()
> - Have set the device name in advance by calling setDeviceName() , even if it's a name given by default